.github/workflows/dev.yaml

```
@@ -1,43 +1,100 @@
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      # To publish to GitHub Pages
+      - name: Publish to GitHub Pages (and render)
+        uses: quarto-dev/quarto-actions/publish@v2
         with:
-          version: 1.7.31
-      - name: Render Quarto Project
-        uses: quarto-dev/quarto-actions/render@v2
+          target: gh-pages
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }} # this secret is always available for github actions
+
       - name: Install npm
         if: ${{ github.event.pull_request.head.repo.full_name == github.repository }}
         uses: actions/setup-node@v2          
         with:
-          node-version: '18'        
-      - name: Deploy to Netlify
-        if: ${{ github.event.pull_request.head.repo.full_name == github.repository }}
-        # NETLIFY_AUTH_TOKEN and NETLIFY_SITE_ID added in the repo's secrets
-        env:
-          NETLIFY_AUTH_TOKEN: ${{ secrets.NETLIFY_AUTH_TOKEN }}
-          NETLIFY_SITE_ID: ${{ secrets.NETLIFY_SITE_ID }}
-          BRANCHE_REF: ${{ github.event.pull_request.head.ref }}
-        run: |
-          npm init -y
-          npm install --unsafe-perm=true netlify-cli -g
-          netlify init
-          netlify deploy --alias=${BRANCHE_REF} --dir="_site" --message "Preview deploy from ${BRANCHE_REF}"
+          node-version: '18'
+
+      - name: Comment PR with preview URL
+        uses: actions/github-script@v6
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
+            const previewUrl = `https://${{ github.repository_owner }}.github.io/${{ github.event.repository.name }}`;
+            github.rest.issues.createComment({
+              issue_number: context.issue.number,
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              body: `✅ Preview site is ready: [${previewUrl}](${previewUrl})`
+            })
+  
+  delete-preview:
+    if: github.event.action == 'closed'  # Only run when PR is closed
+    runs-on: ubuntu-latest
+    steps:
+      - name: Remove preview from GitHub Pages
+        uses: peaceiris/actions-gh-pages@v4.0.0  # https://github.com/marketplace/actions/github-pages-action
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          publish_branch: gh-pages  
+          publish_dir: ./
+          keep_files: false
+          force_orphan: true  # This allows you to make your publish branch with only the latest commit.
+          enable_jekyll: false
+          allow_empty_commit: true
+
+      - name: Comment PR that preview was removed
+        uses: actions/github-script@v6
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
+            github.rest.issues.createComment({
+              issue_number: context.issue.number,
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              body: `🗑️ Preview site has been removed.`
+            })
```

fais gaffe, de manière générale, il faut toujours finir ses fichiers par une ligne vide (ça peut parfois occasionner des heures d'un debug infernal). C'est pour cette raison que GitHub te le signale avec un symbole spécifique.

```suggestion
            })

```